src/@batch-flask/ui/duration-picker/duration-picker.component.ts

```diff
@@ -246,6 +250,18 @@ export class DurationPickerComponent implements FormFieldControl<any>,
                 if (isNaN(time) || time < 0) {
                     this.invalidTimeNumber = true;
                     return null;
+                } else if (time > Constants.MaxDurations.maxDays && this.unit === "days") {
+                    this.invalidDurationValue = true;
+                    return null;
+                } else if (time > Constants.MaxDurations.maxHours && this.unit === "hours") {
+                    this.invalidDurationValue = true;
+                    return null;
+                } else if (time > Constants.MaxDurations.maxMinutes && this.unit === "minutes") {
+                    this.invalidDurationValue = true;
+                    return null;
+                } else if (time > Constants.MaxDurations.maxSeconds && this.unit === "seconds") {
+                    this.invalidDurationValue = true;
+                    return null;
                 } else {
                     return Duration.fromObject({
                         [this.unit]: Number(this.time),
```

Comment:
Actually you should be able to make the constant a Duration matching the actual max duration, then simply compare to the Duration generated here